spec/integration/request_mailer_spec.rb

```diff
@@ -0,0 +1,294 @@
+      expect(tapir_mails.size).to eq(1)
+
+      # Make the followup older
+      outgoing_message.last_sent_at = very_overdue_date
+      outgoing_message.save!
+
+      # Now it should be alerted on
+      RequestMailer.alert_overdue_requests
+      tapir_mails = ActionMailer::Base.deliveries.select{|x| x.body =~ /tapir/}
+      expect(tapir_mails.size).to eq(2)
+    end
+
+  end
+
+  describe "sending unclassified new response reminder alerts" do
+
+    before(:each) do
+      load_raw_emails_data
+    end
+
+    it "sends an alert" do
+      RequestMailer.alert_new_response_reminders
+
+      deliveries = ActionMailer::Base.deliveries
+      expect(deliveries.size).to eq(3) # sufficiently late it sends reminders too
+      mail = deliveries[0]
+      expect(mail.body).to match(/To let everyone know/)
+      expect(mail.to_addrs.first.to_s).to eq(info_requests(:fancy_dog_request).user.email)
+      mail.body.to_s =~ /(http:\/\/.*\/c\/(.*))/
+      mail_url = $1
+      mail_token = $2
+
+      visit mail_url
+      expect(current_path).to match(show_request_path(info_requests(:fancy_dog_request).url_title))
+      # TODO: should check anchor tag here :) that it goes to last new response
+    end
+
+  end
+
+  describe "clarification required alerts" do
+
+    before(:each) do
+      load_raw_emails_data
+    end
+
+    it "should send an alert" do
+      ir = info_requests(:fancy_dog_request)
+      ir.set_described_state('waiting_clarification')
+      # this is pretty horrid, but will do :) need to make it waiting
+      # clarification more than 3 days ago for the alerts to go out.
+      ActiveRecord::Base.connection.update "update info_requests set updated_at = '" + (Time.now - 5.days).strftime("%Y-%m-%d %H:%M:%S") + "' where id = " + ir.id.to_s
+      ir.reload
+
+      RequestMailer.alert_not_clarified_request
+
+      deliveries = ActionMailer::Base.deliveries
+      expect(deliveries.size).to eq(1)
+      mail = deliveries[0]
+      expect(mail.body).to match(/asked you to explain/)
+      expect(mail.to_addrs.first.to_s).to eq(info_requests(:fancy_dog_request).user.email)
+      mail.body.to_s =~ /(http:\/\/.*\/c\/(.*))/
+      mail_url = $1
+      mail_token = $2
+
+      visit mail_url
+      expect(current_path).
+        to match(show_response_path(:id => ir.id,
+                                    :incoming_message_id => ir.incoming_messages.last.id))
+    end
+
+    it "should not send an alert if you are banned" do
+      ir = info_requests(:fancy_dog_request)
+      ir.set_described_state('waiting_clarification')
+
+      ir.user.ban_text = 'Banned'
+      ir.user.save!
+
+      # this is pretty horrid, but will do :) need to make it waiting
+      # clarification more than 3 days ago for the alerts to go out.
+      ActiveRecord::Base.connection.update "update info_requests set updated_at = '" + (Time.now - 5.days).strftime("%Y-%m-%d %H:%M:%S") + "' where id = " + ir.id.to_s
```

Comment:
Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.